cobalt/user_agent/BUILD.gn

```diff
@@ -0,0 +1,45 @@
+# Copyright 2024 The Cobalt Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import("//testing/test.gni")
+
+static_library("user_agent") {
+  sources = [
+    "user_agent_platform_info.cc",
+    "user_agent_platform_info.h",
+  ]
+
+  public_deps = [
+    "//base",
+    "//cobalt:cobalt_build_info",
+    "//starboard/common",
+    "//v8:v8_version",
+  ]
+}
+
+test("user_agent_unittests") {
+  testonly = true
+
+  sources = [
+    "//cobalt/user_agent/user_agent_platform_info_test.cc",
+    "//starboard/common/test_main.cc",
```

Comment:
Usually dependencies like this are not explicit, but via a GN target in `//starboard/common` (something like `test_support`, usually).

If you take my comment above, then either this file or the suggested `//starboard/common:test_support` should be in the deps of `cobalt_unittest` in `cobalt/testing/BUILD.gn`,.

Reply:
I do not find a suitable target for "//starboard/common/test_main.cc", it's included in "//starboard/common:common_test", but that target will pull in many source codes and dependencies not needed for cobalt_unittests, so I will stick to adding the source code.